src/AddressBook.java

```diff
@@ -52,8 +52,13 @@ public void clear() {
         if (persons == null || persons.size() == 0) {
             return;
         }
-        fireTableRowsDeleted(0, persons.size() - 1);
+
+        //Clear persons first
+        int lastRow=persons.size()-1;
         persons.clear();
+
+        //Delete table rows
+        fireTableRowsDeleted(0, lastRow);
```

Comment:
This is what the professor was talking about. I don't know if this is your code or @pvpetit, but when we add, edit, or delete, the professor said it is important to check the correct entry was the one adjusted. Not just one less entry exists. This is what I need to change in my own code.